base/libc.jl

```diff
@@ -170,12 +170,24 @@ library.
 """
 strftime(t) = strftime("%c", t)
 strftime(fmt::AbstractString, t::Real) = strftime(fmt, TmStruct(t))
-function strftime(fmt::AbstractString, tm::TmStruct)
-    timestr = Base.StringVector(128)
-    n = ccall(:strftime, Int, (Ptr{UInt8}, Int, Cstring, Ref{TmStruct}),
-              timestr, length(timestr), fmt, tm)
-    n == 0 && return ""
-    return String(resize!(timestr,n))
+# On Windows use wcsftime instead of strftime to support different locales
+if Sys.iswindows()
+    function wcsftime(fmt::AbstractString, tm::TmStruct)
+        wctimestr = Vector{Cwchar_t}(undef, 128)
+        n = ccall(:wcsftime, Csize_t, (Ptr{Cwchar_t}, Csize_t, Cwstring, Ref{Libc.TmStruct}),
+           wctimestr, length(wctimestr), fmt, tm)
+        n == 0 && return ""
+        return transcode(String, wctimestr[1:n])
+    end
+    strftime(fmt::AbstractString, tm::TmStruct) = wcsftime(fmt, tm)
```

Comment:
Rather than defining both `strftime` and `wcsftime`, you could just call the function `strftime`. Currently you're effectively defining two identical functions on Windows.